File: src/brute_force.cpp
#include <iostream>
using namespace std;

bool isPalindrome(const string& s, int start, int end) {
    while (start < end)
    {
        if (s[start] != s[end]) return false;
        start++;
        end--;
    }
    return true;
}
string longestPalindrome(string s)
{
    int n = s.length();
    for (int len = n; len >= 1; len--)
    {
        for (int start = 0; start + len - 1 < n; start++)
        {
            int end = start + len - 1;
            if (isPalindrome(s, start, end))
            {
                return s.substr(start, len);
            }
        }
    }
    return "";
}
int main()
{
    string input;
    cin >> input;
    cout << longestPalindrome(input) << endl;
}

File: src/brute_force.py
class Brute_Force:

    def is_palindrome(self, s: str, start: int, end: int) -> bool:
        while start < end:
            if s[start] != s[end]:
                return False
            start += 1
            end -= 1
        return True


    def longest_palindrome(self, s : str) -> str:
        n = len(s)
        for length in range(n, 0, -1):  # decreasing substring length
            for start in range(n - length + 1):
                end = start + length - 1
                if self.is_palindrome(s, start, end):
                    return s[start:start + length]
        return ""


if __name__ == "__main__":
    input_str = input()
    print(Brute_Force().longest_palindrome(input_str))
